src/Microsoft.AspNet.Routing/Template/TemplateMatcher.cs

```diff
@@ -36,18 +38,12 @@ public class TemplateMatcher
 
         public RouteTemplate Template { get; private set; }
 
-        public IDictionary<string, object> Match(string requestPath)
+        public IDictionary<string, object> Match(PathString path)
         {
-            if (requestPath == null)
-            {
-                throw new ArgumentNullException(nameof(requestPath));
-            }
-
-            var requestSegments = requestPath.Split(Delimiters);
```

Comment:
The goal of this change is to avoid this call to split, and a few other places where we have to substring.